### PR TITLE
spack recipe for stream with aocc compiler support

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
spack recipe for stream with aocc compiler support and optimization settings for AMD architecture

### CODE REVIEW

File: var/spack/repos/builtin/packages/stream/package.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class Stream(MakefilePackage):
     version('5.10')
 
     variant('openmp', default=False, description='Build with OpenMP support')
+    variant('STREAM_ARRAY_SIZE', default=2500000000, description='Please note that this is system specific setting, please modify it if you want to compile it with different STREAM ARRAY SIZE other than "2500000000". Currently this variant has effect only with AOCC compiler.General recommendation is that STREAM_ARRAY_SIZE must be at least 4x the size of the sum of all the last-level caches in the system')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/stream/package.py:20]:
```suggestion
    variant('STREAM_ARRAY_SIZE', default=2500000000, description='Please note that this is system specific setting, please modify it if you want to compile it with different STREAM ARRAY SIZE other than "2500000000". Currently this variant has effect only with AOCC compiler. General recommendation is that STREAM_ARRAY_SIZE must be at least 4x the size of the sum of all the last-level caches in the system')
```
You may want to break this up on multiple lines as well.

File: var/spack/repos/builtin/packages/stream/package.py [hunk 2]
```diff
@@ -25,9 +26,21 @@ def edit(self, spec, prefix):
         makefile.filter('CC = .*', 'CC = cc')
         makefile.filter('FC = .*', 'FC = f77')
 
-        cflags = '-O2'
-        fflags = '-O2'
-        if '+openmp' in self.spec:
+        if self.compiler.name == 'aocc':
+            omp_flag = self.compiler.openmp_flag
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/stream/package.py:30]: Here you are adding the openmp flag regardless of whether `+openmp` or `~openmp` is requested. I think you should remove the openmp logic from this block and keep it in the subsequent block for all compilers.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/stream/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/stream/package.py:20]: I would also make this lowercase to be consistent with other variant names.

**Comment by amd-toolchain-support** [comment]
With default stream recipe and after passing the flags in the command line, we achieved the desired output.

**Pull request closed without merging**